python/gstaichi/lang/ast/ast_transformer_utils.py

```diff
@@ -25,6 +26,13 @@
     )
 
 
+class PtrSource(Enum):
+    GLOBAL = 1
+    TEMPLATE = 2
+    BUILTIN = 3
+    LOCAL = 4
```

Comment:
Could you add documentation to explain what are all these pointer source types?

Reply:
Yes, I'm in favour of keeping this as simple as possible. If an enum is not required for now, I would remove it for now. Adding it back should not be too difficult since the code path would already be in place.